pkg/envinfo/envinfo_test.go

```diff
@@ -0,0 +1,247 @@
+	os.Setenv(envInfoEnvName, tempEnvFile.Name())
+	testURL := ConfigURL{Name: "testURL", ClusterHost: "1.2.3.4.nip.io", TLSSecret: "testTLSSecret"}
+
+	tests := []struct {
+		name            string
+		parameter       string
+		value           interface{}
+		existingEnvInfo EnvInfo
+	}{
+		{
+			name:      fmt.Sprintf("Case 1: %s to test", URL),
+			parameter: URL,
+			value:     testURL,
+			existingEnvInfo: EnvInfo{
+				componentSettings: ComponentSettings{},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			esi, err := NewEnvSpecificInfo("")
+			if err != nil {
+				t.Error(err)
+			}
+			esi.EnvInfo = tt.existingEnvInfo
+			err = esi.SetConfiguration(tt.parameter, tt.value)
+			if err != nil {
+				t.Error(err)
+			}
+
+			isSet := esi.IsSet(tt.parameter)
+
+			if !isSet {
+				t.Errorf("the '%v' is not set", tt.parameter)
+			}
+
+		})
+	}
+}
+
+func TestLocalUnsetConfiguration(t *testing.T) {
+	tempEnvFile, err := ioutil.TempFile("", "odoenvinfo")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer tempEnvFile.Close()
+	os.Setenv(envInfoEnvName, tempEnvFile.Name())
+	testURL := ConfigURL{Name: "testURL", ClusterHost: "1.2.3.4.nip.io", TLSSecret: "testTLSSecret"}
+
+	tests := []struct {
+		name            string
+		parameter       string
+		value           interface{}
+		existingEnvInfo EnvInfo
+	}{
+		{
+			name:      fmt.Sprintf("Case 1: unset %s", URL),
+			parameter: URL,
+			value:     testURL,
+			existingEnvInfo: EnvInfo{
+				componentSettings: ComponentSettings{
+					URL: &[]ConfigURL{testURL},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			esi, err := NewEnvSpecificInfo("")
+			if err != nil {
+				t.Error(err)
+			}
+			esi.EnvInfo = tt.existingEnvInfo
+
+			err = esi.SetConfiguration(tt.parameter, tt.value)
+			if err != nil {
+				t.Error(err)
+			}
+			isSet := esi.IsSet(tt.parameter)
+			if !isSet {
+				t.Errorf("the '%v' was not set", tt.parameter)
+			}
+
+			err = esi.DeleteEnvSpecificInfo(tt.parameter)
+
+			if err != nil {
+				t.Error(err)
+			}
+			isSet = esi.IsSet(tt.parameter)
+			if isSet {
+				t.Errorf("the '%v' is not set to nil", tt.parameter)
+			}
+
+		})
+	}
+}
+
+func TestLowerCaseParameterForLocalParameters(t *testing.T) {
+	expected := map[string]bool{"url": true}
+	actual := util.GetLowerCaseParameters(GetLocallySupportedParameters())
+	if !reflect.DeepEqual(expected, actual) {
+		t.Errorf("expected '%v', got '%v'", expected, actual)
+	}
+}
+
+func TestLocalConfigInitDoesntCreateLocalOdoFolder(t *testing.T) {
+	// cleaning up old odo files if any
+	filename, err := getEnvInfoFile("")
+	if err != nil {
+		t.Error(err)
+	}
+	os.RemoveAll(filename)
+
+	conf, err := NewEnvSpecificInfo("")
+	if err != nil {
+		t.Errorf("error while creating envinfo %v", err)
+	}
+	if _, err = os.Stat(conf.Filename); !os.IsNotExist(err) {
+		t.Errorf("local env.yaml shouldn't exist yet")
+	}
+}
+
+func TestDeleteConfigDirIfEmpty(t *testing.T) {
```

Comment:
Please name the function name to TestDeleteEnvDirIfEmpty.

Reply:
updated